rollout/canary.go

```diff
@@ -435,6 +435,13 @@ func (c *rolloutContext) reconcileCanaryReplicaSets() (bool, error) {
 		return true, nil
 	}
 
+	// If we have updated both the replica count and the pod template hash c.newRS will be nil we want to reconcile the newRS so we look at the
+	// rollout status to get the newRS to reconcile it.
+	if c.newRS == nil && c.rollout.Status.CurrentPodHash != c.rollout.Status.StableRS {
+		rs, _ := replicasetutil.GetReplicaSetByTemplateHash(c.allRSs, c.rollout.Status.CurrentPodHash)
```

Comment:
Is there a possibility that this function returns the list of RS in the second return param? Is it safe to ignore the returned value?

Reply:
Yes, the other returned variable is a list of Rs we only are interested in the active replicaset which is the first returned value.